service/src/main/kotlin/io/provenance/explorer/service/async/AsyncCaching.kt

```diff
@@ -47,12 +48,12 @@ class AsyncCaching(
 
     protected var chainId: String = ""
 
-    fun getBlock(blockHeight: Int) = getBlockByHeightFromCache(blockHeight)
-
-    fun getBlockByHeightFromCache(blockHeight: Int) = transaction {
+    fun getBlock(blockHeight: Int, checkTxs: Boolean = false) = transaction {
         BlockCacheRecord.findById(blockHeight)?.also {
             BlockCacheRecord.updateHitCount(blockHeight)
-        }?.block ?: saveBlockEtc(blockService.getBlockAtHeightFromChain(blockHeight))
+        }?.block?.also {
+            if (checkTxs) addTxsToCache(it.block.height(), it.block.data.txsCount, it.block.header.time)
```

Comment:
Would it be more efficient to have checkTxs && it.block.data.txCount > 0?

Reply:
Agreed. Updated